src/components/SideNavigation/SideNavigationLink/SideNavigationLink.tsx

```diff
@@ -14,6 +14,10 @@ export type Props<L = LinkDefaultElement, E = HTMLAnchorElement> = Partial<
    * The navigation item's label.
    */
   label: ReactNode;
+  /**
+   * The navigation item's onClick event handler.
+   */
+  onClick?: () => void;
```

Comment:
This looks wrong, Why add a prop, that is being ignored?
Should this be added to `SideNavigationBase` instead and also handled there?

Reply:
When I add forwardRef to the usage in anbox which has onClick prop. I get the error: `Property 'onClick' does not exist on type 'IntrinsicAttributes & Partial<Omit<Props<HTMLSpanElement>, "component">> & { label: ReactNode; component?: ElementType | ... 1 more ... | undefined; forwardRef?: Ref<...> | undefined; }'. Did you mean 'onclick'?`
<img width="1147" alt="image" src="https://github.com/user-attachments/assets/6ce74fae-4b33-45b1-96c9-573e971974dd">

Reply:
Should I add the onClick prop to SideNavBase?
Like this?

```
<Component ref={forwardRef} onClick={onClick} {...props}>
```

I don't get the above error when I add the onClick prop in SideNavBase.

Reply:
These props should get inferred from the component/element type so this suggests a problem with the way `Props` is defined. It looks like changing `Props` to this fixes the issue:

```typescript
export type Props<
  L = LinkDefaultElement,
  E = HTMLAnchorElement,
> = PropsWithSpread<
  {
    /**
     * The navigation item's label.
     */
    label: ReactNode;
    /**
     * The component or element to use for the link element e.g. `a` or `NavLink`.
     * @default a
     */
    component?: SideNavigationBaseProps<L>["component"];
    /**
     * A ref to pass to the link element.
     */
    forwardRef?: React.Ref<E> | null;
  },
  Omit<SideNavigationBaseProps<L>, "component">
>;
```

Then you can remove the `onClick` you added.

Reply:
Yes that's what I was suspecting. Thanks for the fix! It works now.